core/store/src/db/rocksdb.rs

```diff
@@ -415,24 +407,74 @@ impl Database for RocksDB {
         }
     }
 
-    fn create_checkpoint(&self, path: &std::path::Path) -> anyhow::Result<()> {
+    fn create_checkpoint(
+        &self,
+        path: &std::path::Path,
+        columns_to_keep: Option<&[DBCol]>,
+    ) -> anyhow::Result<()> {
         let _span =
             tracing::info_span!(target: "state_snapshot", "create_checkpoint", ?path).entered();
         let cp = ::rocksdb::checkpoint::Checkpoint::new(&self.db)?;
-        cp.create_checkpoint(path)?;
+        cp.create_checkpoint(path)
+            .with_context(|| format!("failed to create checkpoint at {}", path.display()))?;
+
+        if let Some(columns_to_keep) = columns_to_keep {
+            let opts = common_rocksdb_options();
+            let cfs = cf_descriptors(
+                &DBCol::iter().collect::<Vec<_>>(),
+                &StoreConfig::default(),
+                Temperature::Hot,
```

Comment:
That's sus, we'll be using the checkpoint for both hot and rpc databases right? Can you use the same temp as the original db?